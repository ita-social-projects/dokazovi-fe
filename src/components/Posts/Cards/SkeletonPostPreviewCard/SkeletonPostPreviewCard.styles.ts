import { makeStyles, Theme } from '@material-ui/core';

export const useStyles = makeStyles<Theme>(
  (theme: Theme) => ({
    root: {
      position: 'relative',
      borderRadius: 0,
      width: '315px',
    },
    header: () => ({
      display: 'flex',
      height: 182,
      padding: theme.spacing(4, 6, 0, 5),
      marginBottom: theme.spacing(6),
      background: theme.palette.primary.main,
    }),
    body: {
      padding: theme.spacing(0, 6, 0, 5),
    },
    textHeader: {
      wordWrap: 'break-word',
      fontSize: '24px',
      lineHeight: '28px',
      marginBottom: theme.spacing(2.4),
      color: theme.palette.common.black,
    },
    textBody: {
      wordWrap: 'break-word',
      marginBottom: theme.spacing(5),
    },
    postType: {
      color: theme.palette.common.white,
    },
  }),
  {
    name: 'MediaPostPreviewCard',
  },
);
