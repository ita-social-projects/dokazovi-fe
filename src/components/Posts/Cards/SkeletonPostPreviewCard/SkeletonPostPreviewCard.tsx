import React from 'react';
import { Skeleton } from '@material-ui/lab';
import Card from '@material-ui/core/Card';
import Box from '@material-ui/core/Box';
import { useStyles } from './SkeletonPostPreviewCard.styles';

export const SkeletonPostPreviewCard: React.FC = () => {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <Box
        className={classes.header}
        flexDirection="column"
        flexWrap="no-wrap"
        justifyContent="space-between"
      >
        <Skeleton className={classes.postType} component="span" />
      </Box>
      <Box className={classes.body}>
        <Skeleton component="h3" className={classes.textHeader} />
        <Skeleton
          color="textPrimary"
          component="p"
          className={classes.textBody}
        />
        <Box
          display="flex"
          justifyContent="space-between"
          alignItems="center"
          mb={6}
        >
          <Skeleton component="span" color="textSecondary" width="60%" />
        </Box>
      </Box>
    </Card>
  );
};
