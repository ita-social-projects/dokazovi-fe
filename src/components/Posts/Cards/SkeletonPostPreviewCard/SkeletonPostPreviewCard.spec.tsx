import React from 'react';
import renderer from 'react-test-renderer';
import { SkeletonPostPreviewCard } from './SkeletonPostPreviewCard';

describe('SkeletonPostPreviewCard', () => {
  it('renders the component', () => {
    const tree = renderer.create(<SkeletonPostPreviewCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
