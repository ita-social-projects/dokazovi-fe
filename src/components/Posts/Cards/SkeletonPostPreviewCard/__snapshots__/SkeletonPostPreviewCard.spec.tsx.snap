// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonPostPreviewCard renders the component 1`] = `
<div
  className="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="skeleton"
>
  <div
    className="MuiBox-root MuiBox-root-8 MediaPostPreviewCard-header-2 MediaPostPreviewCard-header-7"
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-6 MuiSkeleton-pulse"
      style={
        Object {
          "height": undefined,
          "width": undefined,
        }
      }
    />
  </div>
  <div
    className="MuiBox-root MuiBox-root-9 MediaPostPreviewCard-body-3"
  >
    <h3
      className="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-4 MuiSkeleton-pulse"
      style={
        Object {
          "height": undefined,
          "width": undefined,
        }
      }
    />
    <p
      className="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-5 MuiSkeleton-pulse"
      color="textPrimary"
      style={
        Object {
          "height": undefined,
          "width": undefined,
        }
      }
    />
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        color="textSecondary"
        style={
          Object {
            "height": undefined,
            "width": "60%",
          }
        }
      />
    </div>
  </div>
</div>
`;
